goas/atomic_asset_transfer/common/src/lib.rs

```diff
@@ -170,3 +182,18 @@ pub enum Input {
     Withdraw(Withdraw),
     Deposit(Deposit),
 }
+
+impl Input {
```

Comment:
What do you think of renaming `Input` to `Tx` or `ZoneTx`? we call these inputs "transactions" everywhere else